File: appointments-service/src/internal/internal.controller.ts
import {
  BadRequestException,
  Controller,
  Get,
  Query,
  UseGuards,
} from '@nestjs/common';
import { InternalService } from './internal.service';
import { InternalApiKeyGuard } from './internal-api-key.guard';

@UseGuards(InternalApiKeyGuard)
@Controller('internal')
export class InternalController {
  constructor(private readonly internalService: InternalService) {}

  @Get('relationships/patient-dentist')
  checkPatientDentistRelation(
    @Query('patientId') patientId: string,
    @Query('dentistId') dentistId: string,
  ) {
    if (!patientId || !dentistId) {
      throw new BadRequestException('patientId and dentistId are required');
    }

    return this.internalService.hasPatientDentistRelation(patientId, dentistId);
  }
}

File: appointments-service-csharp/Dentia.Appointments.Api/Controllers/InternalController.cs
using Dentia.Appointments.Api.Application.Common;
using Dentia.Appointments.Api.Application.Services;
using Microsoft.AspNetCore.Mvc;

namespace Dentia.Appointments.Api.Controllers;

[ApiController]
[Route("internal")]
public class InternalController : ControllerBase
{
    private readonly IAppointmentsService _appointmentsService;
    private readonly IConfiguration _configuration;

    public InternalController(
        IAppointmentsService appointmentsService,
        IConfiguration configuration)
    {
        _appointmentsService = appointmentsService;
        _configuration = configuration;
    }

    [HttpGet("relationships/patient-dentist")]
    public async Task<IActionResult> CheckPatientDentistRelation(
        [FromQuery] string patientId,
        [FromQuery] string dentistId)
    {
        var apiKey = Request.Headers["x-internal-api-key"].FirstOrDefault();
        var expectedApiKey = _configuration["INTERNAL_API_KEY"] ?? "dev-internal-key";

        if (string.IsNullOrWhiteSpace(apiKey) || apiKey != expectedApiKey)
        {
            throw new AppException(StatusCodes.Status401Unauthorized, "Invalid internal API key");
        }

        if (string.IsNullOrWhiteSpace(patientId) || string.IsNullOrWhiteSpace(dentistId))
        {
            throw new AppException(StatusCodes.Status400BadRequest, "patientId and dentistId are required");
        }

        var result = await _appointmentsService.HasPatientDentistRelationAsync(patientId, dentistId);
        return Ok(result);
    }
}